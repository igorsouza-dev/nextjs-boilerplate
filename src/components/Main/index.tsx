import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="React avancado" />
    <S.Title>Nextjs boilerplate</S.Title>
    <S.Description>
      Typescript, ReactJS, NextJS and Styled Components
    </S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="developer in front of a computer"
    />
  </S.Wrapper>
)

export default Main
