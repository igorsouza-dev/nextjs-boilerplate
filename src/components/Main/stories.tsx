import React from 'react'

import { Story } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main
}

const Template: Story = (args) => <Main {...args} />

export const Basic = Template.bind({})

Basic.args = {
  title: 'Boilerplate',
  description: 'Nextjs'
}
