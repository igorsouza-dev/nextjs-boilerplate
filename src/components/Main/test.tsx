import { render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('Should render the heading', () => {
    const { container, getByRole } = render(<Main />)
    const heading = getByRole('heading', { name: 'advanced react' })
    expect(heading).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
