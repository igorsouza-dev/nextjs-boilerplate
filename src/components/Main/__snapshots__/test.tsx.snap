// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render the heading 1`] = `
<div>
  <main
    class="styles__Wrapper-heyFP kMDFpc"
  >
    <img
      alt="React avancado"
      class="styles__Logo-bSryDj imcKrg"
      src="/img/logo.svg"
    />
    <h1
      class="styles__Title-cLvJfs hkqwnH"
    >
      Nextjs boilerplate
    </h1>
    <h2
      class="styles__Description-elWOBK gJKaYL"
    >
      Typescript, ReactJS, NextJS and Styled Components
    </h2>
    <img
      alt="developer in front of a computer"
      class="styles__Illustration-fyJZJM fjqpFV"
      src="/img/hero-illustration.svg"
    />
  </main>
</div>
`
